From: John Feeney <jfeeney@redhat.com>
Subject: [RHEL-5 PATCH] Built-in speaker on Dell Mobile Precision (Fila) with  STAC9205 codec
Date: Thu, 21 Dec 2006 13:42:14 -0500
Bugzilla: 219494
Message-Id: <458AD586.8050104@redhat.com>
Changelog: sound: add support for STAC9205 codec


bugzilla 219494
Built-in speaker on Dell Mobile Precision (Fila) with STAC9205 codec
doesn't work
http://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=219494

   /At Fri, 15 Dec 2006 16:08:59 -0600,/
   / Matt Porter wrote:/
   / > /
   / > Adds support for handling EAPD on 9205 codecs/
   / > /
   / > Signed-off-by: Matt Porter <mporter@em...>/

   / Applied to ALSA HG tree. Thanks./

   /Takashi /

diff -u linux-2.6.18.i386/sound/pci/hda/patch_sigmatel.c linux-2.6.18.i386/sound/pci/hda/patch_sigmatel.c
--- linux-2.6.18.i386/sound/pci/hda/patch_sigmatel.c	2006-12-07 16:20:23.000000000 -0500
+++ linux-2.6.18.i386/sound/pci/hda/patch_sigmatel.c	2006-12-07 16:57:06.000000000 -0500
@@ -1678,6 +1678,15 @@
 
 	spec->multiout.dac_nids = spec->dac_nids;
 
+	/* Configure GPIO0 as output */
+	snd_hda_codec_write(codec, codec->afg, 0, AC_VERB_SET_GPIO_DIRECTION, 0x00000001);
+	/* Configure GPIO0 as CMOS */
+	snd_hda_codec_write(codec, codec->afg, 0, 0x7e7, 0x00000000);
+	/* Assert GPIO0 high */
+	snd_hda_codec_write(codec, codec->afg, 0, AC_VERB_SET_GPIO_DATA, 0x00000001);
+	/* Enable GPIO0 */
+	snd_hda_codec_write(codec, codec->afg, 0, AC_VERB_SET_GPIO_MASK, 0x00000001);
+
 	err = stac92xx_parse_auto_config(codec, 0x1f, 0x20);
 	if (err < 0) {
 		stac92xx_free(codec);

